src/mscorlib/src/System/Diagnostics/Eventing/XplatEventLogger.cs

```diff
@@ -109,24 +109,53 @@ private static string Serialize(ReadOnlyCollection<string> payloadName, ReadOnly
                 sb.Append("\\\"");
                 sb.Append(':');
 
-                var valuestr = payload[i] as string;
-
-                if( valuestr != null)
-                {
-                    sb.Append("\\\"");
-                    minimalJsonserializer(valuestr,sb);
-                    sb.Append("\\\"");
-                }
-                else
+                switch(payload[i])
                 {
-                    sb.Append(payload[i].ToString());
+                    case string str:
+                    {
+                        sb.Append("\\\"");
+                        minimalJsonserializer(str, sb);
+                        sb.Append("\\\"");
+                        break;
+                    }
+                    case byte[] byteArr:
+                    {
+                        sb.Append("\\\"");
+                        AppendByteArrayAsHexString(sb, byteArr);
+                        sb.Append("\\\"");
+                        break;
+                    }
+                    default:
+                    {
+                        sb.Append(payload[i].ToString());
+                        break;
+                    }
                 }
-
             }
             sb.Append('}');
             return StringBuilderCache.GetStringAndRelease(sb);
         }
 
+        private static void AppendByteArrayAsHexString(StringBuilder builder, byte[] byteArray)
+        {
+            Debug.Assert(builder != null);
+
+            // Treat NULL byte arrays as empty byte arrays.
+            if(byteArray == null)
```

Comment:
is this case possible? the payload is an `ReadOnlyCollection<object>`, in case of switch(payload[i]) the null case is going to be handled by `default` case

```cs
ReadOnlyCollection<object> payload = new ReadOnlyCollection<object>(new object[] { "string", new byte[] { 1 }, null });

for (int i = 0; i < payload.Count; i++)
	switch (payload[i])
	{
		case string str:
		{
			Console.WriteLine("string");
			break;
		}
		case byte[] byteArr:
		{
			Console.WriteLine("byteArr");
			break;
		}
		default:
		{
			Console.WriteLine("default");
			break;
		}
	}
```

Prints:

string
byteArr
default

Reply:
Doesn't look like it.  I've converted the byte[] null check to an assert, and have also wrapped the default case in a null check since null.ToString() is going to be a problem...